packages/microsite/src/cli/microsite-dev.tsx

```diff
@@ -247,17 +249,32 @@ export default async function dev(
       if (base.endsWith(".html")) base = base.slice(0, ".html".length * -1);
       if (base === "") base = "index";
 
-      const loadAndSSR = async (base: string) => {
-        try {
-          const url = `/src/pages/${base}.js`;
-          const result = await snowpack.loadUrl(url, { isSSR: true });
-          if (!result) throw new Error();
+      const findPageComponentPathForBaseUrl = async (base: string): Promise<string|null> => {
```

Comment:
One more thing I was wondering about is whether writing explicit return types is acceptable. I prefer to use them as they increase readability and make code less error-prone. However, most functions in this file rely on inferring so you might want to make it consistent if you have an opinion on this =)